src/components/PostDetails.tsx

```diff
@@ -2,125 +2,74 @@ import React, { useEffect, useState } from 'react';
-  const [loaded, setLoaded] = useState(false);
-  const [hasError, setError] = useState(false);
+import { CommentData } from '../types/Comment';
+import { useAppDispatch, useAppSelector } from '../app/hooks';
+import {
+  addComments,
+  deleteComments,
+  fetchComments,
+} from '../features/thunks/commentsThunk';
+import { commentsActions } from '../features/slices/commentsSlice';
+
+export const PostDetails: React.FC = () => {
+  const dispatch = useAppDispatch();
+  const { comments, error, loading } = useAppSelector(state => state.comments);
+  const { currentPost: post } = useAppSelector(state => state.posts);
   const [visible, setVisible] = useState(false);
 
-  function loadComments() {
-    setLoaded(false);
-    setError(false);
-    setVisible(false);
-
-    commentsApi.getPostComments(post.id)
-      .then(setComments) // save the loaded comments
-      .catch(() => setError(true)) // show an error when something went wrong
-      .finally(() => setLoaded(true)); // hide the spinner
-  }
-
-  useEffect(loadComments, [post.id]);
-
-  // The same useEffect with async/await
-  /*
-  async function loadComments() {
-    setLoaded(false);
+  useEffect(() => {
     setVisible(false);
-    setError(false);
-
-    try {
-      const commentsFromServer = await commentsApi.getPostComments(post.id);
-
-      setComments(commentsFromServer);
-    } catch (error) {
-      setError(true);
-    } finally {
-      setLoaded(true);
+    if (post?.id !== undefined) {
+      dispatch(fetchComments(post.id));
     }
-  };
-
-  useEffect(() => {
-    loadComments();
-  }, []);
-
-  useEffect(loadComments, [post.id]); // Wrong!
-  // effect can return only a function but not a Promise
-  */
+  }, [dispatch, post?.id]);
 
   const addComment = async ({ name, email, body }: CommentData) => {
-    try {
-      const newComment = await commentsApi.createComment({
-        name,
-        email,
-        body,
-        postId: post.id,
-      });
-
-      setComments(
-        currentComments => [...currentComments, newComment],
-      );
-
-      // setComments([...comments, newComment]);
-      // works wrong if we wrap `addComment` with `useCallback`
-      // because it takes the `comments` cached during the first render
-      // not the actual ones
-    } catch (error) {
-      // we show an error message in case of any error
-      setError(true);
-    }
+    const data = {
+      name,
+      email,
+      body,
+      postId: post?.id ?? 0,
+    };
+
+    await dispatch(addComments(data));
   };
 
   const deleteComment = async (commentId: number) => {
-    // we delete the comment immediately so as
-    // not to make the user wait long for the actual deletion
-    setComments(
-      currentComments => currentComments.filter(
-        comment => comment.id !== commentId,
-      ),
-    );
-
-    await commentsApi.deleteComment(commentId);
+    dispatch(commentsActions.removeComment(commentId));
+    await dispatch(deleteComments(commentId));
```

Comment:
no need to call both - you have extraReducer and you handle this case here:
![image](https://github.com/mate-academy/react_redux-list-of-posts/assets/86923089/155f5b80-6168-407a-8e02-b23d4ca65b1f)

Also, no need to use `await` here